src/NServiceBus.Core/Pipeline/PipelineComponent.cs

```diff
@@ -24,19 +25,31 @@ public static PipelineComponent Initialize(PipelineSettings settings, HostingCom
                 step.ApplyContainerRegistration(hostingConfiguration.Container);
             }
 
-            return new PipelineComponent(modifications);
+            return new PipelineComponent(settings);
         }
 
         public Pipeline<T> CreatePipeline<T>(IBuilder builder) where T : IBehaviorContext
         {
-            return new Pipeline<T>(builder, modifications);
+            return new Pipeline<T>(builder, pipelineSettings.modifications);
         }
 
         public PipelineCache BuildPipelineCache(IBuilder rootBuilder)
         {
-            return new PipelineCache(rootBuilder, modifications);
+            return new PipelineCache(rootBuilder, pipelineSettings.modifications);
         }
 
-        readonly PipelineModifications modifications;
+        public void RegisterBehavior<T>(string stepId, Func<IBuilder, T> factoryMethod, string description)
```

Comment:
I think that's not a valid approach. Behaviors registered with the settings need to be registered with the container which is currently done in `Initialize`. If you register stuff on the component, the container registrations are missing?